### ISSUE TITLE
Publish different transpilation targets to different npm packages

### ISSUE THREAD

**Issue by jeffbcross** [ReactiveX/RxJS#751]
In order to have nicer import paths, without `dist` and module/language directories, we should publish the different outputs (CJS, ES6, TS) as different npm packages: rxjs for CJS, rxjs-es6 for ES6, rxjs-ts for TypeScript. It would also be nice to drop the `@reactivex` scoping in the process, since that causes issues for users with older npm versions.

Before:
```typescript
import {map} from '@reactivex/rxjs/dist/cjs/operators/map';
```

After (though #750 will move the operators even further up the export tree)
```typescript
import {map} from 'rxjs/operators/map';
```

**Comment by robwormald** [comment]
i can do this if you like @jeffbcross

**Comment by robwormald** [comment]
i dont think we need the TS distro either - just distribute the `.d.ts` files alongside the ES6 and CJS output (like we do already)

**Comment by blesh** [comment]
Perhaps it should be `rxjs-es` ... because some day it will be `es7` or `es2025` or whatever. But hopefully by then the module wars will be over.

**Comment by blesh** [comment]
... then again that could imply `es3` haha. I give up. What makes sense?

**Comment by robwormald** [comment]
SystemJS et al refer to it as `esm` - https://github.com/systemjs/systemjs/blob/master/docs/module-formats.md

**Comment by trxcllnt** [comment]
What about `rxjs-node`, `rxjs` and `rxjs-ts` for CJS, ES6+, and TS respectively?

**Comment by blesh** [comment]
@trxcllnt: I discussed this on Slack with @jeffbcross, and he brought up a good point on the `rxjs-node` front, which is that CJS isn't really node specific. Many developers use CJS for their client-side bundling processes.

**Comment by jeffbcross** [comment]
I also like the idea of CJS being the default, non-suffixed (rxjs) distribution, since it is in the current native format of node modules (CJS).

**Comment by jeffbcross** [comment]
Closed by @robwormald in 185abdfb24065d014263ebabfb0af5d5fda3fb42

(Issues have been opened to fix bugs introduced in the new process)
